src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperationPage.java

```diff
@@ -2,21 +2,73 @@
 
 import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
 import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
 import io.stargate.sgv2.jsonapi.service.shredding.model.DocumentId;
 import io.stargate.sgv2.jsonapi.service.shredding.model.WritableShreddedDocument;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Supplier;
 
 /**
- * The internal to modification operation results, what were the ID's of the docs we changed and
- * what change.
+ * The internal to insert operation results, keeping ids of successfully and not-successfully
+ * inserted documents.
+ *
+ * <p>Can serve as an aggregator, using the {@link #aggregate(WritableShreddedDocument, Throwable)}
+ * function.
+ *
+ * @param insertedIds Documents IDs that we successfully inserted.
+ * @param failedIds Document IDs that failed to be inserted.
  */
 public record InsertOperationPage(
-    List<DocumentId> insertedIds, List<WritableShreddedDocument> insertedDocs)
+    List<DocumentId> insertedIds, Map<DocumentId, Throwable> failedIds)
     implements Supplier<CommandResult> {
+
+  /** No-arg constructor, usually used for aggregation. */
+  public InsertOperationPage() {
+    this(new ArrayList<>(), new HashMap<>());
+  }
+
+  /** {@inheritDoc} */
   @Override
   public CommandResult get() {
+    // if we have errors, transform
+    if (null != failedIds && !failedIds().isEmpty()) {
+
+      List<CommandResult.Error> errors = new ArrayList<>(failedIds.size());
+      failedIds.forEach((documentId, throwable) -> errors.add(getError(documentId, throwable)));
```

Comment:
currently for each failed document, I return an error, should we aggregate? if yes how?